lib/main.dart

```diff
@@ -23,42 +26,102 @@ class MyApp extends StatelessWidget {
+  Widget build(BuildContext context, WidgetRef ref) {
     return Scaffold(
       appBar: AppBar(),
       body: Center(
         child: Column(
           mainAxisAlignment: MainAxisAlignment.center,
           children: <Widget>[
             const Text(
-              'You have pushed the button this many times:',
+              'Minus:',
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                ElevatedButton(
+                    onPressed: () {
+                      ref.read(counterNotifierProvider.notifier).increment();
+                    },
+                    child: const Text("Boutton plus")),
+                Consumer(
+                  builder: (context, ref, _) {
+                    final minus = ref.watch(counterNotifierProvider);
+                    return Text(
+                      '$minus',
+                      style: Theme.of(context).textTheme.headlineMedium,
+                    );
+                  },
+                ),
+                ElevatedButton(
+                    onPressed: () {
+                      ref.read(counterNotifierProvider.notifier).decrement();
+                    },
+                    child: const Text("Boutton moin")),
+              ],
+            ),
+            const Gap(20),
+            const Text(
+              'Addition:',
             ),
-            Consumer(
-              builder: (BuildContext context, WidgetRef ref, _) {
-                final counter = ref.watch(counterNotifierProvider);
-                return Text(
-                  '$counter',
-                  style: Theme.of(context).textTheme.headlineMedium,
-                );
-              },
+            Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                ElevatedButton(
+                    onPressed: () {
+                      ref.read(additionNotifierProvider.notifier).increment();
+                    },
+                    child: const Text("Boutton plus")),
+                Consumer(
+                  builder: (context, ref, _) {
+                    final addition = ref.watch(additionNotifierProvider);
+                    return Text(
+                      '$addition',
+                      style: Theme.of(context).textTheme.headlineMedium,
+                    );
+                  },
+                ),
+                ElevatedButton(
+                    onPressed: () {
+                      ref.read(additionNotifierProvider.notifier).decrement();
+                    },
+                    child: const Text("Boutton moin")),
+              ],
+            ),
+            const Gap(20),
+            const Text(
+              'Opération de Minus & Addition:',
+            ),
+            Consumer(builder: (context, ref, _) {
+              final add = ref.watch(getAdditionProvider);
+              return Text("$add");
+            }),
```

Comment:
Comme ton widget est un `ConsumerWidget` tu n'as plus besoin d'encapsuler dans un `Consumer` ! 
Tu peux faire 

```suggestion
             Text("${ref.watch(getAdditionProvider)}"),
```